Source/ASCollectionView.mm

```diff
@@ -805,6 +809,31 @@ - (void)invalidateFlowLayoutDelegateMetrics
   // Subclass hook
 }
 
+- (nullable NSString *)modelIdentifierForElementAtIndexPath:(NSIndexPath *)indexPath inView:(UIView *)view {
+    if (_asyncDataSourceFlags.modelIdentifierMethods) {
+        GET_COLLECTIONNODE_OR_RETURN(collectionNode, nil);
+        NSIndexPath *convertedPath = [self convertIndexPathToCollectionNode:indexPath];
+        if (convertedPath == nil) {
+            return nil;
+        } else {
+            return [_asyncDataSource modelIdentifierForElementAtIndexPath:convertedPath inNode:collectionNode];
+        }
+    } else {
+        return nil;
+    }
+}
+
+- (nullable NSIndexPath *)indexPathForElementWithModelIdentifier:(NSString *)identifier inView:(UIView *)view {
+    if (_asyncDataSourceFlags.modelIdentifierMethods) {
+        GET_COLLECTIONNODE_OR_RETURN(collectionNode, nil);
+        NSIndexPath *result = [_asyncDataSource indexPathForElementWithModelIdentifier:identifier inNode:collectionNode];
+        result = [self convertIndexPathToCollectionNode:result];
```

Comment:
The data source should return us an index path in the collection node space, so we can return without converting.

Reply:
Thanks! I've made this change on both `ASTableView` and `ASCollectionView`

Reply:
Excellent!